CONTRIBUTING.md

```diff
@@ -40,14 +39,204 @@ If you happen to have a fix for a package, please have a look at the
+documentation. A minimal example is as follows:
+
+```
+$ cat CHANGES
+
+## v0.0.1 (2021-02-23)
+### Added
+- Meta files for dune release
+```
+
+After updating the CHANGES file, you need to tag the release. You will
+be prompted to confirm the version, as can be seen in the following
+output:
+
+```
+$ dune-release tag
+[-] Extracting tag from first entry in CHANGES
+[-] Using tag "v0.0.1"
+[?] Create git tag v0.0.1 for HEAD? [Y/n]
+Y
+[+] Tagged HEAD with version v0.0.1
+```
+
+* `dune-release distrib`
+
+The sources are then archived to create a distribution as illustrated
+below:
+
+```
+ $ dune-release distrib
+[-] Building source archive
+[+] Wrote archive _build/hello-v0.0.1.tbz
+
+[-] Linting distrib in _build/hello-v0.0.1
+[ OK ] File README is present.
+[ OK ] File LICENSE is present.
+[ OK ] File CHANGES is present.
+[ OK ] File opam is present.
+[ OK ] lint opam file hello.opam.
+[ OK ] opam field description is present
+[ OK ] opam fields homepage and dev-repo can be parsed by dune-release
+[ OK ] Skipping doc field linting, no doc field found
+[ OK ] lint _build/hello-v0.0.1 success
+
+[-] Building package in _build/hello-v0.0.1
+Done: 56/60 (jobs: 1)[ OK ] package builds
+
+[-] Running package tests in _build/hello-v0.0.1
+Done: 0/0 (jobs: 0)[ OK ] package tests
+
+[+] Distribution for hello 0.0.1
+[+] Commit c9935ac42271ce23cc422c299c5c3aaf92646499
+[+] Archive _build/hello-v0.0.1.tbz
+```
+
+* `dune-release publish`
+
+You can now publish the distribution to your GitHub repository as
+follows:
+
+```
+$ dune-release publish
+[-] Skipping documentation publication for package hello: no doc field in hello.opam
+[-] Publishing distribution
+[-] Publishing to github
+[?] Push tag v0.0.1 to git@github.com:guest/hello? [Y/n]
+Y
+[-] Pushing tag v0.0.1 to git@github.com:guest/hello
+[?] Create release v0.0.1 on git@github.com:guest/hello? [Y/n]
+Y
+[-] Creating release v0.0.1 on git@github.com:guest/hello via github's API
+[+] Succesfully created release with id 38465706
+[?] Upload _build/hello-v0.0.1.tbz as release asset? [Y/n]
+Y
+[-] Uploading _build/hello-v0.0.1.tbz as a release asset for v0.0.1 via github's API
+```
+
+* `dune-release opam pkg`
+
+The .opam file can then be generated using the following command:
+
+```
+$ dune-release opam pkg
+[-] Creating opam package description for hello
+[+] Wrote opam package description _build/hello.0.0.1/opam
+```
+
+* `dune-release opam submit`
+
+Before you can submit the package to opam.ocaml.org, you need to first
+fork `github.com/ocaml/opam-repository` to your GitHub account, and
+also clone your copy locally to your system.
+
+```
+$ git clone git@github.com:guest/opam-repository
+```
+
+The `~/.config/dune/release.yml` file should then be updated with your
+opam repository settings as indicated below:
+
+```
+$ cat ~/.config/dune/release.yml
+
+user: guest
+remote: git@github.com:guest/opam-repository
+local: /home/guest/opam-repository/
```

Comment:
````suggestion
The `~/.config/dune/release.yml` file should then be updated with your
opam repository settings as indicated below (in particular, change `<username>` by your local username and `<github-user>` by your own github username):

```
$ cat ~/.config/dune/release.yml

user: <username>
remote: git@github.com:<github-username>/opam-repository.git
local: /home/<username>/.cache/dune/opam-repository/
```
````
I think it is better to explicit what to change. I also changed the path of the local repository to be in a more user friendly place. See: https://github.com/ocamllabs/dune-release/issues/286